### PR TITLE
`@0x/contracts-integrations`: Restructure directories, welcome core.ts and balance stores 

### PR DESCRIPTION
**Pull request by mzhu25**
## Description
Moves a bunch of files and directories around in the integrations package. The updated structure:
Screen Shot 2019-11-13 at 13 43 40
I'd potentially like to have balance stores use stuff from the integrations test framework (DeploymentManager, actors, simulations, etc.), which required moving balance stores out of the exchange package. 
This required also moving `core.ts` and `FillOrderWrapper`. This in turn required moving some signature validation tests from `core` to `signature_validation.ts`. 🙈 And I also deleted `TestSignatureValidator.sol` while I was there.
I didn't do a whole refactor of `core.ts` in this PR, just wanted to get the move done. 

Also changes a bunch of imports to be directly from the file instead of one-off modules, per #2330, and brings us one file closer to killing `ExchangeWrapper` 🎊 

## Checklist:

-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Update documentation as needed.
-   [ ] Add new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: contracts/integrations/test/framework/assertions/function_assertion.ts [hunk 1]
```diff
@@ -106,79 +106,3 @@ export class FunctionAssertion<TBefore, ReturnDataType> implements Assertion {
         };
     }
 }
-
-export type IndexGenerator = () => number;
```
Comment (jalextowle) [inline contracts/integrations/test/framework/assertions/function_assertion.ts]: RIP. Can't believe we never even tried these.

File: contracts/integrations/test/exchange/core_test.ts [hunk 2]
```diff
@@ -249,115 +232,16 @@ blockchainTests.resets('Exchange core', () => {
             signedOrder = await orderFactory.newSignedOrderAsync();
         });
 
-        describe('callback signature types', () => {
```
Comment (jalextowle) [inline contracts/integrations/test/exchange/core_test.ts]: Why were these removed?

Comment (mzhu25) [reply contracts/integrations/test/exchange/core_test.ts]: they were moved to `signature_validator.ts` in `@0x/contracts-exchange` because they rely on `TestValidatorWalletContract`

File: contracts/test-utils/src/types.ts [hunk 3]
```diff
@@ -201,3 +201,14 @@ export enum ExchangeFunctionName {
     SetProtocolFeeCollectorAddress = 'setProtocolFeeCollectorAddress',
     DetachProtocolFeeCollector = 'detachProtocolFeeCollector',
 }
+
+export enum ValidatorWalletAction {
```
Comment (dorothy-zbornak) [inline contracts/test-utils/src/types.ts]: Do we use these anywhere other than the signature validator tests?

File: contracts/exchange/test/signature_validator.ts [hunk 4]
```diff
@@ -1118,21 +1118,168 @@ blockchainTests.resets('MixinSignatureValidator', env => {
...
+                ExchangeSignatureValidatorApprovalEventArgs
             >;
             const logArgs = log.args;
             expect(logArgs.signerAddress).to.equal(signerAddress);
             expect(logArgs.validatorAddress).to.equal(validatorWallet.address);
             expect(logArgs.isApproved).to.equal(approval);
         });
     });
+
+    describe('fillOrder integration tests', () => {
```
[the remark this answers is not in the document]
Comment (mzhu25) [reply contracts/exchange/test/signature_validator.ts]: yeah these are pretty gross but the cases in `isolated_fill_order.ts` use different signatures, whereas these use the same signature but change the validator wallet's state ☹️ 
If we ever do a tech debt demolition sprint I've got a list of files to delet or refactor and this'll be one of them

File: contracts/test-utils/src/types.ts (hunk 3 above)
Comment (mzhu25) [reply contracts/test-utils/src/types.ts]: nope, good catch
will move
